azure/datalake/store/core.py

```diff
@@ -107,73 +107,103 @@ def open(self, path, mode='rb', blocksize=2**25, delimiter=None):
         return AzureDLFile(self, AzureDLPath(path), mode, blocksize=blocksize,
                            delimiter=delimiter)
 
-    def _ls(self, path):
+    def _ls(self, path, invalidate_cache=True):
         """ List files at given path """
         path = AzureDLPath(path).trim()
         key = path.as_posix()
+        
+        if invalidate_cache:
+            self.invalidate_cache(key)
```

Comment:
Yes, because our service is a distributed filesystem being touched by many users across many different machines, it doesn't make sense to default to using the cached copy unless the user explicitly wants to and understands the risks of "staleness".